homeassistant/components/diagnostics/__init__.py

```diff
@@ -23,7 +24,7 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
@frenck I think changes to `diagnostics` should trigger full CI.
For this PR I don't think it's necessary but maybe for the future?

Reply:
Yes, it should, it is a core feature that needs to be added to the core files file.

Reply:
PR opened for the CI: #64417